Create getting started page for CHI@EDGE (#57)
ChameleonCloud/docs: source/getting-started-edge/index.rst

.. _InCommon: https://incommon.org/federation

.. _getting-started:

=============================
Getting started with CHI@EDGE
=============================

This guide will walk you through the initial steps of getting an account,
joining a project and working with your first container.

> .. contents:: :local:

Reviewer: What does this do?

Author: This generates a TOC for the local page, based on the headers, which is also on the normal getting started page. Here is a screenshot:
![image](https://user-images.githubusercontent.com/9397092/120835359-d6064c00-c529-11eb-8930-a32ec258d14e.png)

.. _getting-started-user:

Step 1: Log in to Chameleon
===========================

> Just click the “Log in” button situated in the top right corner of `our main page

Reviewer: Do these smart quotes render OK?

Reviewer: (I'm guessing this was copied, so it must!)

Author: They render as the same character smart quotes, nothing is converted.

If you do not meet these criteria (e.g., students generally do not), you will
need to ask your advisor or other scientist supervising your research to create
the project for you. You can request PI status by checking a box in `your
Chameleon profile <https://www.chameleoncloud.org/user/profile/>`_. Chameleon PI
status requests are typically reviewed within one business day.

Once you have PI status, you may apply for a new project with an initial
allocation. A project application typically consists of a short description of
your intended research and takes one business day to process. Once your project
has been approved, you will be able to utilize the testbed sites.

Step 3: Start using Chameleon!
==============================

Congratulations, you are now ready to launch your first container! Containers
are a simple way to deploy applications. Learn more about what containers are 
with `this guide from docker <https://www.docker.com/resources/what-container>`_.
Follow these steps to launch a container and manage it.

The CHI\@EDGE dashboard
----------------------

Chameleon edge resources are available through 

> `CHI@EDGE <https://chi.edge.chameleoncloud.org>`_. When you access this site, you are 

Reviewer: ```suggestion
`CHI@Edge <https://chi.edge.chameleoncloud.org>`_. When you access this site, you are 
```

We'll probably use "Edge" as it is not an acronym. But it would be funny to imagine an acronym for EDGE. "Edge Device General Environment"? Hehe

   you already have a security group here that meets your needs, you can skip 
   the next step. Otherwise, you will need to create a new security group.

#. To create a new security group, click *+ Create Security Group*, enter a name 
   for your new group in the wizard, and then click *Create Security Group*. 
   You will be redirected to the screen to manage your new group's rules. Select 
   *Add Rule* to open the add rule wizard. Under *Rule*, you can select from 
   common rules, or if your needs are not met by one of these options, select 
   one of the custom rules. For example, if your container runs a web
   server, you may want to add the rules HTTP and HTTPS, allowing for traffic
   on ports 80 and 443.

  .. figure:: create_security_group.png
     :alt: Create Security Group Wizard
     :figclass: screenshot

     Enter a name for the security group.

Launching a container
---------------------

To start launching a container, follow the following steps:

1. In the sidebar, click *Container*, then click *Containers*.

2. Click on the *Create Container* button in the toolbar and the *Create
   Container* wizard will load

> 3. Type *my_first_container* for the container name. Then, enter the name of an 

Reviewer: It may be good to be explicit that the container name can be anything. 

"Give your container a name. For example, since it's your first container, "my_first_container" might be a good name."

   image you want to launch from Docker Hub. You must use the full name of the 
   image. Optionally, you can supply a custom command to override the default
   command run by Docker.

  .. note::
    You may also use a Glance ID for your image, by selecting *Glance* under the

>     *Image Driver* field.

Reviewer: ```suggestion
    *Image Driver* field. You should only provide a Glance ID if you've previously created a container snapshot, which stores the snapshot as a Glance image.
```

Reviewer: (Feel free to adjust that english, it's a bit clunky perhaps)

>   .. note::

Reviewer: ```suggestion
  .. important::
```

Reviewer: (there are various levels of notices, this one is quite important)

    Only the ARM architecture is currently supported. Make sure the image used 

>     is compatible with ARM.

Reviewer: ```suggestion
    is compatible with ARM. `Here is a list of such images on Docker Hub <https://hub.docker.com/search?type=image&architecture=arm64>`_.
```

  .. figure:: create_container.png
    :alt: Create Container wizard
    :figclass: screenshot

     Enter a name and image name.

4. Click *Networks* in sidebar. Then, find *containernet1* in the image list and
   click the *Up* arrow to select it.

   .. figure:: create_container_networks.png
      :alt: Selecting a network
      :figclass: screenshot

      Select the containernet1 network

5. Click *Security Groups* in sidebar. Select the security group you wish to use
   by clicking the *Up* arrow to select it.

   .. figure:: create_container_security_groups.png
      :alt: Select security groups to use
      :figclass: screenshot

      You select your desired security group.

6. Click the *Create* button.

Congratulations, you have launched an container! It may take a few minutes for

> your container to become active, while the image downloads.

Reviewer: ```suggestion
your container to become active, if the image is not yet downloaded to the target device.
```

Just making it a bit more clear that it's not always going to be that way.
